openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -233,32 +233,49 @@ private Schema handleTypeReference(Map<String, Schema> schema, TypeReferenceType
      *     };
      * </pre>
      */
-    private Schema handleUnionType(Map<String, Schema> schema, Map.Entry<String, RecordFieldSymbol> field,
-                                   Schema property) {
-
-        List<TypeSymbol> unionTypes = ((UnionTypeSymbol) field.getValue().typeDescriptor()).userSpecifiedMemberTypes();
-        String componentName = null;
+    private Schema handleUnionType(UnionTypeSymbol unionType, Schema property) {
+        List<TypeSymbol> unionTypes = unionType.userSpecifiedMemberTypes();
+        //Set array size to 4 by assuming union type can have max 4 types.
+        List<Schema> properties = new ArrayList<>(4);
```

Comment:
Are there any specific reasons for this assumption?

Reply:
Since by default, it takes list array size 10, due to this avoiding the memory allocation.

Reply:
I see. I was concerned since theres no compile time / runtime guarantee that a union type cannot have more than 4 nested types AFAIK